src/runtime/internal/utils.ts

```diff
@@ -96,7 +96,9 @@ export let now: () => number = is_client
 	? () => window.performance.now()
 	: () => Date.now();
 
-export let raf = is_client ? requestAnimationFrame : noop;
+export let raf = (callback) => is_client && window.requestAnimationFrame
+	? window.requestAnimationFrame.apply(window, callback)
```

Comment:
This would need to be `.call`, not `.apply`. `.apply` expects an array of arguments as its second argument.

Reply:
It could also just export a bound version, not doing anything with arguments

```js
export let raf = is_client ? requestAnimationFrame.bind(window) : noop;
```

Reply:
Oh right, nice, that does seem better.

Reply:
@lukeed I tried that, but the problem was the tests started failing when I did, so I went with this instead.
<img width="1642" alt="Screenshot 2019-06-13 at 20 07 32" src="https://user-images.githubusercontent.com/5371454/59456569-e988eb80-8e16-11e9-884e-790e8b19ae6c.png">

However @Conduitry is correct, I should be using call here. Silly mistake.

Reply:
Peculiar that it's throwing an error about not being defined. At least the way I've written it, its scope hasn't changed. Leads me to think that changes elsewhere here caused the change in scope

Reply:
For the tests to work with that version `requestAnimationFrame` would need to be added to `global`, as JSDOM doesn't do that by default.

We can't take that route however as property access hints side effects at bundlers and `raf` wouldn't be tree-shakeable then.

Reply:
I think those `is not defined` errors are just an artifact of how the test setup is written currently. But even if we worked around those, `requestAnimationFrom.bind(window)` would still fail to be treeshaken out by Rollup.

Reply:
Curious, would `/*@__PURE__*/` fix that here? re: `.bind(window)` shaking

Reply:
Possibly. But Rich had a concern (see comments on #2963) about this. At the very least, that wouldn't work with people using versions of Rollup <1.3.0.

Reply:
Yeah, I'm not sure what tools support that pure annotation these days. At least Rollup and Terser do, but I have no idea how the more unconventional bundlers deal with that.

Is it easy to get Rollup to keep the annotation in the output?

Reply:
Rollup does leave the annotations in by default I am pretty sure. Are you not seeing them in the output? It could be someone else in the build process stripping comments.

If we went the annotation route, we'd need to also update Agadoo, which currently is using an old version of Rollup to test for treeshakability, and which does not support the pure annotations.